### PR TITLE
feat: add create new resource option for folder item

### PR DESCRIPTION
**Pull request by erdkse**
This PR...

## Changes

- Adds a new option to file items in treeview to create a new resource

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/FileTreePane/FileTreePane.tsx [hunk 1]
```diff
@@ -372,6 +375,8 @@ const TreeItem: React.FC<TreeItemProps> = props => {
     ? fileMap[ROOT_FILE_ENTRY].filePath
     : path.join(fileMap[ROOT_FILE_ENTRY].filePath, treeKey);
 
+  const targetFolder = isRoot ? ROOT_FILE_ENTRY : treeKey.replace(path.sep, '');
```
Comment (olensmar) [inline src/components/organisms/FileTreePane/FileTreePane.tsx:378]: hm.. I think targetFolder should be relative to root - or empty for root!? Does this work for nested folders?

Comment (erdkse) [reply src/components/organisms/FileTreePane/FileTreePane.tsx:378]: It creates files in the nested folders but can't be expanded. (might not be related)

I followed the structure of the folder dropdown in NewResourceWizard.

**Comment by olensmar** [comment]
looks good although I noticed that new resource files are not shown in the file-explorer - maybe they don't get added to the filemap!? or the fileexplorer doesn't update on changes?

**Comment by erdkse** [comment]
I fixed that - actually there is a seamless way to solve it and tried for that but gave up because lack of time - will continue working on it

File: src/models/ui.ts [hunk 2]
```diff
@@ -95,6 +95,7 @@ export type UiState = {
   resetLayout: boolean;
   clusterPaneIconHighlighted: boolean;
   clusterStatusHidden: boolean;
+  reloadFileTree: boolean;
```
Comment (olensmar) [inline src/models/ui.ts:98]: hm... why is this necessary? doesn't the file-tree react to changes in the fileMap ?

Comment (erdkse) [reply src/models/ui.ts:98]: It does - but there is another problem I added this as temporary

**Review by olensmar — APPROVED** [review]

**Comment by olensmar** [comment]
nice work @erdkse !
